hrabsence/CRM/HRAbsence/BAO/HRAbsenceType.php

```diff
@@ -117,6 +77,59 @@ public static function create($params) {
   }
 
   /**
+   * Creates Activitu Types based on $params and $activityTypes
+   * @param  $activityTypes Array
+   * @param  $params        Array
+   * @param  $dr_cr         Boolean 'true' for 'debit' and 'false' for 'credit'
+   * @return Array          Created Activity Type as array
+   */
+  public static function createActivityType($activityTypes, $params, $dr_cr)
+  {
+    $weight = count($activityTypes["values"]);
+    if ($dr_cr) {
+      $valueId = $params['debit_activity_type_id'];
+      $valueLabel = $params['title'];
+    } else {
+      $valueId = $params['credit_activity_type_id'];
+      $valueLabel = ts('%1 (Credit)', array(1 => $params["title"]));
+    }
+    $paramsCreate = array(
+      'weight' => ($weight+1),
+      'label' => $valueLabel,
+      'filter' => 1,
+      'is_active' => 1,
+      'is_optgroup' => 0,
+      'is_default' => 0,
+      'grouping' => 'Timesheet'
+    );
+    return civicrm_api3('activity_type', 'create', $paramsCreate);
+  }
+
+  /**
+   * Updates Opton Value based on $params
+   * @param  $params  Array
+   * @param  $dr_cr   Boolean  'true' for 'debit' and 'false' for 'credit'
+   */
+  public static function updateOptionValue($params, $dr_cr) {
+    if ($dr_cr) {
+      $valueId = $params['debit_activity_type_id'];
+      $valueLabel = $params['title'];
+    } else {
+      $valueId = $params['credit_activity_type_id'];
+      $valueLabel = ts('%1 (Credit)', array(1 => $params["title"]));
+    }
+    $optionValue = civicrm_api3('OptionValue', 'get', array(
+      'sequential' => 1,
+      'option_group_id' => "activity_type",
+      'value' => $valueId
+    ));
+    civicrm_api3('OptionValue', 'create', array(
+      'id' => $optionValue['id'],
+      'label' => $valueLabel
+    ));
```

Comment:
Please always try to add some spaces to make the code easier to read. You could add on empty line after the if/else block and one between the civicrm_api3 calls